HTTP 403 error while executing `pre.sh`

rustls-webpki uses this action to install cargo-check-external-types:

```
Run taiki-e/cache-cargo-install-action@v2
Run set -eu
Run bash --noprofile --norc "${GITHUB_ACTION_PATH:?}/pre.sh"
  bash --noprofile --norc "${GITHUB_ACTION_PATH:?}/pre.sh"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    CARGO_HOME: /home/runner/.cargo
    CARGO_INCREMENTAL: 0
    CARGO_TERM_COLOR: always
    INPUT_TOOL: cargo-check-external-types
    INPUT_LOCKED: true
    INPUT_GIT: 
    INPUT_TAG: 
    INPUT_REV: 
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
curl: (22) The requested URL returned error: 403
Error: Process completed with exit code 22.
```

https://github.com/rustls/webpki/actions/runs/15729589302/job/44327351500?pr=359

--- a/action.yml
+++ b/action.yml
@@ -59,6 +59,7 @@ runs:
         INPUT_GIT: ${{ inputs.git }}
         INPUT_TAG: ${{ inputs.tag }}
         INPUT_REV: ${{ inputs.rev }}
+        ACTION_USER_AGENT: ${{ github.action_repository }} (${{ github.action_ref }})
     - name: Restore Cache
       id: cache
       uses: actions/cache@v4

--- a/pre.sh
+++ b/pre.sh
@@ -265,7 +265,7 @@ if [[ "${version}" == "latest" ]] || [[ -n "${fetch}" ]]; then
     *) bail "unsupported host OS '${host_os}'" ;;
   esac
 
-  crate_info=$(retry curl --proto '=https' --tlsv1.2 -fsSL --retry 10 "https://crates.io/api/v1/crates/${tool}")
+  crate_info=$(retry curl -v --user-agent "${ACTION_USER_AGENT}" --proto '=https' --tlsv1.2 -fsSL --retry 10 "https://crates.io/api/v1/crates/${tool}")
   case "${version}" in
     latest)
       version=$(jq -r '.crate.max_stable_version' <<<"${crate_info}")

